instrumentation/opentelemetry-instrumentation-pymongo/src/opentelemetry/instrumentation/pymongo/__init__.py

```diff
@@ -226,6 +226,13 @@ def _get_statement_by_command_name(
         return statement
 
 
+def _get_command_collection_name(event: CommandEvent) -> str | None:
+    collection_name = event.command.get(event.command_name)
+    if not collection_name or not isinstance(collection_name, str):
+        return None
```

Comment:
I think this makes sense since the semconv for [db.collection.name](https://opentelemetry.io/docs/specs/semconv/registry/attributes/db/#db-collection-name) (replacement for [DB_MONGODB_COLLECTION](https://github.com/open-telemetry/opentelemetry-python/blob/5ddb8e7407a06b560c5483f7649913e170229936/opentelemetry-semantic-conventions/src/opentelemetry/semconv/_incubating/attributes/db_attributes.py#L164-L167)) uses the "recommended" wording instead of "required". If None then `set_attribute` is not called at all.